### DISCUSSION TITLE
"Hide top bar and move close, minimize, restore buttons to top left " CSS snippet stopped working after update

### DISCUSSION THREAD

**Discussion by coreyr2** [piroor/treestyletab#3677] [General]
hi, the "Hide top bar and move close, minimize, restore buttons to top left (MacOS Big Sur, Firefox 89+)" snippet doesnt work for mac anymore. any help appreciated <333

btw heres my userchrome file rn:

```
#main-window[tabsintitlebar="true"]:not([extradragspace="true"]) #TabsToolbar > .toolbar-items {
  opacity: 0;
  pointer-events: none;
}
#main-window:not([tabsintitlebar="true"]) #TabsToolbar {
    visibility: collapse !important;
}
#main-window[tabsintitlebar="true"]:not([extradragspace="true"]) #TabsToolbar .titlebar-spacer {
        border-inline-end: none;
}

#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] #sidebar-header {
  display: none;
}

#titlebar {
  appearance: none !important;
  height: 0px;
}

#titlebar > #toolbar-menubar {
  margin-top: 0px;
}

#TabsToolbar {
  min-width: 0 !important;
  min-height: 0 !important;
}

#TabsToolbar > .titlebar-buttonbox-container {
  display: block;
  position: absolute;
  top: 12px;
  left: 0px;
}

#tabbrowser-tabs {
visibility: collapse !important;
min-height: 0 !important;
}

tab{
display:none!important;
}
#main-window #TabsToolbar {
   display: none;
}  

#titlebar {
  appearance: none !important;
  height: 0px;
}

#titlebar > #toolbar-menubar {
  margin-top: 0px;
}

#TabsToolbar {
  min-width: 0 !important;
  min-height: 0 !important;
}

#TabsToolbar > .titlebar-buttonbox-container {
  display: block;
  position: absolute;
  top: 12px;
  left: 0px;
}

```

**Comment by piroor** [comment]
By the way you can see the native styling of the title bar with no horizontal tab bar with setting both `sidebar.revamp` and `sidebar.verticalTabs` to `true` on recent versions of Firefox. Setting `sidebar.visibility` to `hide-sidebar` allows you to hide the vertical bar (new style sidebar header), so it may be a solution better than fragile userChrome.css.

**Reply by piroor** [reply]
They correspond to options UI in Nightly:

* Options => Firefox Labs => Turn on "Sidebar" and "Vertical tabs"
* Click the "gear" button in the vertical bar => "Sidebar button" => Choose "Show and hide sidebar"

**Reply by coreyr2** [reply]
i guess that works but now i have the new vertical tabs along with the tst tabs. i cant hide the vertical tabs without it also turning off tst

**Comment by iunodora** [comment]
I found a workaround that, while not perfect, is a good enough solution while we wait for TST to update their CSS recommendations post Firefox v133.

The solution: change the #TabsToolbar section to the following:
```
#TabsToolbar {
  max-width: 0 !important;
  max-height: 0 !important;
  min-height: 0 !important;
}
```
This is the result:
![image](https://github.com/user-attachments/assets/c1dff929-1940-4bf6-8d24-fc7be9ed7f73)


The only downside I found so far is that making the window very narrow results in the close, minimize, restore buttons covering the stuff immediately to the right. But who uses a squashed window like that anyway ^^ Here's the view of a super duper narrow window:
![image](https://github.com/user-attachments/assets/5e138019-72da-471d-9ebb-7cda441f869d)
